pkg/controller/combo_controller.go

```diff
@@ -2,36 +2,158 @@ package controller
 
 import (
 	"context"
+	"fmt"
+	"strings"
+	"time"
 
 	"github.com/go-logr/logr"
+	"github.com/operator-framework/combo/api/v1alpha1"
+	combinationPkg "github.com/operator-framework/combo/pkg/combination"
+	comboConditions "github.com/operator-framework/combo/pkg/conditions"
+	templatePkg "github.com/operator-framework/combo/pkg/template"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
 
-	"github.com/operator-framework/combo/api/v1alpha1"
+var (
+	RequeueDefaultTime = time.Second * 2
+	SuccessCondition   = comboConditions.ProccessedCombinationsCondition
 )
 
 type combinationController struct {
 	client.Client
 	log logr.Logger
 }
 
+// manageWith creates a new instance of this controller
 func (c *combinationController) manageWith(mgr ctrl.Manager, version int) error {
 	c.log = c.log.V(version)
+	templateHandler := handler.EnqueueRequestsFromMapFunc(c.mapTemplateToCombinations)
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.Combination{}).
+		Watches(&source.Kind{Type: &v1alpha1.Template{}}, templateHandler).
 		Complete(c)
 }
 
+func (c *combinationController) mapTemplateToCombinations(obj client.Object) []reconcile.Request {
+	if obj == nil {
+		return nil
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	templateName := obj.GetName()
+
+	// Find all of the combinations that rely on this template and enqueue them for updates
+	var requests []reconcile.Request
+	combinationList := v1alpha1.CombinationList{}
+	if err := c.List(ctx, &combinationList, &client.ListOptions{Namespace: obj.GetNamespace()}); err != nil {
+		c.log.V(0).Error(err, "error when listing combinatons: ")
+		return requests
+	}
+
+	for _, combination := range combinationList.Items {
+		if combination.Spec.Template == templateName {
+			c.log.Info(fmt.Sprintf("enqueueing %s combination in response to associated %s template being updated", combination.Name, templateName))
```

Comment:
nit but this doesn't have to be addressed here: let's bump up that log level at some point.

Reply:
Let's address that in this issue:

https://github.com/operator-framework/combo/issues/62